### PR TITLE
fix: cloud init problems with ubuntu

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:

This pr attempts to solve: 

https://jira.nutanix.com/browse/NCN-100790
https://jira.nutanix.com/browse/NCN-102002
 
by only using our copied datasource module on older versions of cloud-init. this aligns us a little closer to the [upstream image-builder tasks](https://github.com/kubernetes-sigs/image-builder/blob/main/images/capi/ansible/roles/providers/tasks/vmware-redhat.yml) 

attempting to build these images but im unable to run kib locally with my credentials for vsphere

this is safe for the current version we are pinning in ubuntu -- [23.1.2](https://github.com/mesosphere/konvoy-image-builder/blob/main/bundles/ubuntu20.04/bundle.sh.gotmpl#L25) because it already contains [DataSourceVmWare.py](https://github.com/canonical/cloud-init/blob/ubuntu/23.1.2-0ubuntu0_20.04.1/cloudinit/sources/DataSourceVMware.py) 

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by faiq** [comment]
I dont think this is quite ready 

 this somehow pulled in an even older version of cloud init.

**Comment by faiq** [comment]
```
root@e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8:/home/konvoy# crictl ps
CONTAINER           IMAGE               CREATED              STATE               NAME                      ATTEMPT             POD ID              POD
003b714215ceb       3af2ab51e1364       About a minute ago   Running             kube-apiserver            0                   733aa6dcf4429       kube-apiserver-e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8
2610995525975       3861cfcd7c04c       About a minute ago   Running             etcd                      0                   a2e1573adb1f9       etcd-e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8
32cfb42264efb       083b81fc09e85       About a minute ago   Running             kube-controller-manager   0                   f4e61bc2f8b05       kube-controller-manager-e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8
ca9e900c8fb6d       49d9b8328a8fd       About a minute ago   Running             kube-scheduler            0                   3a7f747eb0957       kube-scheduler-e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8
root@e2e-vsphere-air-gapped-test-1724250617-control-plane-6q2h8:/home/konvoy# cloud-init --version
/usr/bin/cloud-init 24.1.3-0ubuntu1~20.04.5
``` 

looks like this is working fine - added some extra commits to make it pull a newer cloud init.

File: ansible/roles/providers/tasks/vmware.yml [hunk 1]
```diff
@@ -25,6 +25,14 @@
       - cloud-initramfs-dyn-netconf
   when: ansible_os_family == "Debian"
 
+- name: Ensure cloud-init version > 23.1.3 is installed
+  apt:
+    name: "cloud-init=24.1.3-0ubuntu1~20.04.5"
```
Comment (jimmidyson) [inline ansible/roles/providers/tasks/vmware.yml:30]: This pins the version, right? Is that what we want to do here or allow > 23.1.3?

Comment (faiq) [reply ansible/roles/providers/tasks/vmware.yml:30]: i added this commit to sort of force the situation that the reported bugs came in -

without this commit i was not able to get a new version of cloud init for some reason. 

We can keep it here, if we want to force a newer version

**Comment by faiq** [comment]
for some reason i get v22.0 but that has datasource.py 

https://github.com/canonical/cloud-init/blob/ubuntu/22.2-0ubuntu1_20.04.1/cloudinit/sources/DataSourceVMware.py


```
vsphere-clone.kib_image: TASK [providers : Set cloud-init version] **************************************
    vsphere-clone.kib_image: ok: [default]
    vsphere-clone.kib_image:
    vsphere-clone.kib_image: TASK [providers : Deprecation warning for disabling password authentication] ***
    vsphere-clone.kib_image: ok: [default] => {
    vsphere-clone.kib_image:     "msg": "22.2-0ubuntu1~20.04.3"
    vsphere-clone.kib_image: }
    vsphere-clone.kib_image:
    vsphere-clone.kib_image: TASK [providers : Install cloud-init packages] *********************************
    vsphere-clone.kib_image: skipping: [default]
 ```
 
 i think this is happening because our base iso is old

File: ansible/roles/providers/tasks/vmware.yml [hunk 2]
```diff
@@ -12,6 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 ---
+# FIXME: remove after upgrading ansible
```
Comment (faiq) [inline ansible/roles/providers/tasks/vmware.yml:15]: to do remove this `

File: .github/workflows/release-vsphere-template.yaml [hunk 3]
```diff
@@ -22,7 +22,7 @@ jobs:
           - os: "redhat 8.8"
             buildConfig: "offline-fips"
           - os: "ubuntu 20.04"
-            buildConfig: "offline"
```
Comment (supershal) [inline .github/workflows/release-vsphere-template.yaml:25]: lets add both basic and offline tests.

File: ansible/roles/providers/tasks/vmware.yml (hunk 2 above)
Comment (supershal) [reply ansible/roles/providers/tasks/vmware.yml]: removed it.

**Comment by supershal** [comment]
added https://github.com/mesosphere/konvoy-image-builder/pull/1150/commits/9b114a754545f5bd258cb371983d14f1c9ca7160 to remove cloud-init dropin that configures cloud-init to use custom vmware guest datasource.

**Comment by supershal** [comment]
Created release template and running konvoy e2e tests.
RHEL 8.5 https://github.com/mesosphere/konvoy2/actions/runs/10498724952/job/29084248822 🟢 
Ubuntu 2204: https://github.com/mesosphere/konvoy2/actions/runs/10498700886 (running)

**Review by supershal — APPROVED** [review]

**Pull request merged**